src/sentry_json.c

```diff
@@ -220,8 +221,16 @@ sentry__jsonwriter_write_double(sentry_jsonwriter_t *jw, double val)
     if (can_write_item(jw)) {
         char buf[50];
         // The MAX_SAFE_INTEGER is 9007199254740991, which has 16 digits
-        sentry__snprintf_c(buf, sizeof(buf), "%.16g", val);
-        write_str(jw, buf);
+        size_t written
+            = (size_t)sentry__snprintf_c(buf, sizeof(buf), "%.16g", val);
+        // print `null` if we have printf issues or a non-finite double, which
+        // can't be represented in JSON.
```

Comment:
doesn't this mean that it's impossible to create json adhering to some schema?  any consumer now has to expect null.  Might be better to error out entirely?

Reply:
That is also true. Problem is that the codepaths right now are infallible (void functions), which means this is the only sensible things to do which at least does not produce outright invalid JSON.

Reply:
Yeah, but at least that's consistent with what we do in other SDKs.